stac_fastapi/types/stac_fastapi/types/search.py

```diff
@@ -56,19 +120,71 @@ def kwargs(self) -> Dict:
 class BaseSearchGetRequest(APIRequest):
     """Base arguments for GET Request."""
 
-    collections: Annotated[Optional[str], Query()] = attr.ib(
-        default=None, converter=str2list
+    collections: Optional[List[str]] = attr.ib(
+        default=None, converter=_collection_converter
     )
```

Comment:
we can now defined the type correctly (because it's not the type used in the Query parameter, but the type returned by the converter)

ref #682